Manual Gimbap Migration

Ap0c opened the pull request:
Migrated the Manual Gimbap from frontend to native template.

![gimbap-compare](https://cloud.githubusercontent.com/assets/5131341/18675347/fbb8516c-7f49-11e6-87b2-1bc5475b37a1.png)

- Tested in **preview** and in **test ad unit** on the main site.
- Matches the original at all breakpoints, and for both **1x1x1x1** and **2x1x1** layouts:

![gimbap2x1x1-compare](https://cloud.githubusercontent.com/assets/5131341/18677393/88ac45fa-7f50-11e6-89e3-c71bee24257c.png)

- Tested in Safari, Firefox, Chrome and Edge. Works in IE11, but with some components cut out, I think this may be intentional:

![ie-gimbap](https://cloud.githubusercontent.com/assets/5131341/18676492/b00e49e8-7f4d-11e6-922d-9530167cc20a.png)

- Have not been able to test with author images yet.

@regiskuckaertz

regiskuckaertz commented on build/manual-gimbap/index.html:
```diff
@@ -0,0 +1,100 @@
+</aside>
+
+<script>
+var quoteSvg = `<span class="gimbap__quote inline-quote inline-icon"><svg aria-hidden="true" class="icon icon--quote" width="17" height="10" viewBox="0 0 33 20"><path d="M9.002 20c3.85.068 6.932-3.104 7-6.994.068-3.892-3.15-6.937-7-7.006-2.016-.036-3.59.694-4.888 2.053-.07-.48-.122-1.04-.113-1.553C4.064 2.97 7.414.937 11.003 1l-1-1C3.98 0 .098 4.447.002 10.006c-.097 5.557 3.35 9.892 9 9.994zM26.0 ...
```
do you need the surrounding `SPAN`?

regiskuckaertz commented on build/manual-gimbap/index.html:
```diff
@@ -0,0 +1,100 @@
+</svg>
+`
+};
+var authorImage = '[%Offer1AuthorImage%]';
```
This can go in `index.js`, as well as the line below.

regiskuckaertz commented on src/manual-gimbap/index.js:
```diff
@@ -0,0 +1,46 @@
+    getWebfonts();
+});
+
+// Adds quotes to title if set in DFP.
```
All this can be wrapped in a call to `write` (see in `_shared/js/dom.js`) to perform DOM mutations inside the animation frame callback.

regiskuckaertz commented:
Also, the back ticks are ES6 syntax, we need to figure out a way to get rid of them in the index.html file, because they won't be transpiled.

Ap0c commented:
Changes made:

- Moved the DFP variables that had no need to be in `index.html` to the js file.
- Tweaked the build script to strip newlines from SVGs, obviating the need for template literals in html file.
- Rewrote the `index.js` file to make use of `requestAnimationFrame` via the `write` method.
- Left the spans wrapping the SVGs, as they are in the original template and have styling attached to them.

regiskuckaertz commented on lib/html.js:
```diff
@@ -3,11 +3,15 @@ const hogan = require('hogan.js');
-                return svg.replace('<svg ', `<svg aria-hidden="true" class="icon icon--${svgName}" `);
+                return svg
+                    .replace('<svg ', `<svg aria-hidden="true" class="icon icon--${svgName}" `)
+                    .replace(/\r?\n|\r/g, '');
```
Well done, mate!
